src/main/java/com/ita/if103java/ims/service/impl/UserServiceImpl.java

```diff
@@ -107,16 +108,21 @@ public UserDto findAdminByAccountId(Long accountID) {
         return mapper.toDto(userDao.findAdminByAccountId(accountID));
     }
 
+    @Transactional
     @Override
     public UserDto update(UserDto userDto) {
-        User updatedUser = mapper.toEntity(userDto);
-        //Activating status can't be changed in this way
-        User dbUser = userDao.findById(updatedUser.getId());
-        updatedUser.setActive(dbUser.isActive());
-        updatedUser.setPassword(passwordEncoder.encode(userDto.getPassword()));
+        User user = userDao.findById(userDto.getId());
+        user.setFirstName(userDto.getFirstName());
+        user.setLastName(userDto.getLastName());
+        user.setPassword(passwordEncoder.encode(userDto.getPassword()));
 
-        User user = userDao.update(updatedUser);
+        User updatedUser = userDao.update(user);
+        user.setUpdatedDate(updatedUser.getUpdatedDate());
         eventService.create(createEvent(user, PROFILE_CHANGED, "updated profile."));
+
+        AccountDto account = new AccountDto();
+        account.setName(userDto.getAccountName());
+        accountService.update(user, account);
```

Comment:
hm, do we need to have this here? 
We added this for create, but for update we should have separate page, isn't it? 
@DashaBoichuk please review too

Reply:
I think that when the user will want to edit profile info, he should have ability to change an account name also, hasn't he?

Reply:
I don't think so. We are going have separate page for Account settings with ability to update name

Reply:
hmm, okay) Let it be.

Reply:
@ikrasWork I think we can give possibility to admin to change his account name from both places. From account-view page and from his profile settings. It will be comfortably for him. What do you think about that?

Reply:
up to you. For me it is strange, however if you decided to allow this, please go ahead.